### PR TITLE
Better mapbox type definition and layer labeling

### PR DESCRIPTION
**Pull request by frodrigo**
* Better Mapbox type definition, expression start with as array.
* Use more generic let to name mapbox expression block

### CODE REVIEW

File: src/icon.ts [hunk 1]
```diff
@@ -7,8 +7,8 @@ enum StyleName {
 
 interface Styles {
   beforeLayer?: string;
-  imageLayout?: MapboxValues;
-  textColor?: MapboxValues;
+  imageLayout?: MapboxExpr;
+  textColor?: MapboxExpr;
```
Comment (GuillaumeAmat) [inline src/icon.ts:11]: This is wrong, `textColor` is not an array, and it was not a `MapboxValues` actually... So just a `string`?

File: src/icon.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ export function icon(map: mapboxgl.Map, options?: Options) {
       imageLayout: map.getLayoutProperty('poi-level-1', 'icon-image'),
       textColor: map.getPaintProperty('poi-level-1', 'text-color'),
       ...options?.styles?.[StyleName.Icon],
-    },
+    } as Styles,
```
Comment (GuillaumeAmat) [inline src/icon.ts:50]: I'd rather remove both `as Styles` and type the `const styles`:

```
const styles: { [key in StyleName]: Styles } = ...
```

File: types/index.d.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
 declare var __DEV__: boolean;
 
-type MapboxValues = undefined | string | number | boolean | null | MapboxValues[];
+type MapboxValues = undefined | string | number | boolean | null | MapboxExpr;
```
Comment (GuillaumeAmat) [inline types/index.d.ts:3]: It feels weird to have two types like that, I'd rather refact the former one like that:

```
type MapboxValues = Array<undefined | string | number | boolean | null | MapboxValues>;
```

File: src/icon.ts (hunk 1 above)
Comment (frodrigo) [reply src/icon.ts:11]: `imageLayout` and `textColor` should `MapboxValues`, not `MapboxExpr`.
Fixed.

File: types/index.d.ts (hunk 3 above)
Comment (frodrigo) [reply types/index.d.ts:3]: We need both (`textColor`).

File: src/icon.ts (hunk 2 above)
Comment (frodrigo) [reply src/icon.ts]: Done.

**Comment by frodrigo** [comment]
Note, I will remove icon.ts in next PR.

**Comment by frodrigo** [comment]
It merged, but the commit is not on master.

**Comment by frodrigo** [comment]
Manually added on master.

**Comment by GuillaumeAmat** [comment]
Weird, Github lost the commit somehow...

**Pull request merged**